DRILL-6751: Upgrade Apache parent POM to version 21

vdiravka opened the pull request:
- Update apache.pom file version to 21 (with updating some maven plugins versions)
- Include Drill's sources jars on assembly stage in <moduleSets> (not <dependencySets>)
  for properincluding jars with last apache-21.pom
- Separate "distro-assembly" to the two execution stages to avoid:
  [WARNING] Assembly file: ___ is not a regular file (it may be a directory).
  It cannot be attached to the project build for installation or deployment.
- Remove unsused <include>/<exclude> in assebly descriptor to avoid:
  [WARNING] The following patterns were never triggered in this artifact inclusion filter
- Update "maven-assembly-plugin" version
- Update "slf4j" version
- Update "avatica version
- Update "mockito-core" version
- Update "bcpkix-jdk15on" (Bouncy Castle Cryptography APIs) version

vdiravka commented:
@sohami Could you please review?

sohami commented on distribution/src/assemble/component.xml:
```diff
@@ -18,42 +18,127 @@
+        <fileSets>
+          <fileSet>
+            <directory>target</directory>
+            <outputDirectory>/jars</outputDirectory>
```
should this be `jars` only instead of `/jars` as specified in `outputDirectory` of `dependencySet` ?

sohami commented on exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/WrappedAccessor.java:
```diff
@@ -327,6 +328,19 @@ public Array getArray() throws SQLException {
+      if (message != null && message.startsWith("cannot convert to")) {
+        throw new SQLException(e.getMessage(), e);
+      }
+      throw e;
```
Looks like the error handling can be refactored to a separate method and used everywhere.

sohami commented on pom.xml:
```diff
@@ -1679,28 +1681,20 @@
-                  <runOnlyAtExecutionRoot>true</runOnlyAtExecutionRoot>
                   <appendAssemblyId>false</appendAssemblyId>
-                  <descriptors>
-                    <descriptor>src/main/resources/assemblies/source-assembly.xml</descriptor>
```
How are we making sure that excludes defined in source-assembly.xml are still taken care of ?

vdiravka replied:
I have manually checked that all files are the same which obtained in the result of `mvn clean install -DskipTests -Papache-release`.
Looks like there are no differences, except those mentioned in Laurent's comment in the PR #449

vdiravka commented:
@sohami Thank you for review.

Looks like there is no more need in converting `RuntimeException` to `SQLException`, it was resolved in Avatica-1.9.0 for a long time ago, see [CALCITE-1408](https://issues.apache.org/jira/browse/CALCITE-1408)
For now it is [`SQLDataException`](https://github.com/apache/calcite-avatica/blob/4b7eee5bf430b916c7c07897b6f60d2b6b6dabb7/core/src/main/java/org/apache/calcite/avatica/util/AbstractCursor.java#L367) (inheritor of `java.sql.SQLException`).
Therefore no need to have WrappedAccessor in Drill.

I have updated it and squashed "Changes after the review" with the last commit.

sohami commented:
@vdiravka - Thanks for catching it and making the change. Looks like there is build failure due to unused imports. Please fix that.

```
[ERROR] /home/travis/build/apache/drill/exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java:38:8: Unused import - java.util.ArrayList. [UnusedImports]
[ERROR] /home/travis/build/apache/drill/exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java:39:8: Unused import - java.util.List. [UnusedImports]
[ERROR] /home/travis/build/apache/drill/exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java:50:8: Unused import - org.apache.calcite.avatica.util.Cursor.Accessor. [UnusedImports]
```

sohami commented:
+1
